kubernetes.go

```diff
@@ -169,6 +195,18 @@ func existGatewayCRDs(ctx context.Context, c *gatewayClient.Clientset) bool {
 	panic(err)
 }
 
+func existVirtualServerCRDs(ctx context.Context, c *k8s_nginx.Clientset) bool {
+	_, err := c.K8sV1().VirtualServers("").List(ctx, metav1.ListOptions{})
+	if meta.IsNoMatchError(err) || runtime.IsNotRegisteredError(err) || errors.IsNotFound(err) {
+		log.Infof("VirtualServer CRDs are not found. Not syncing VirtualServer resources.")
+		return false
+	}
```

Comment:
can you update this function with https://github.com/ori-edge/k8s_gateway/pull/71?
can you also see if it's possible to factor out this entire error check into a single function?

Reply:
Updated.
I quickly looked at combining the functions, but the key problem is that each uses a different implementation of `ClientSet`. We could pass both `ClientSet` objects into a single function and return an ordered set of bools to cover each CRD.
Would that be acceptable?

Reply:
I was thinking more along the lines of the error processing function...
so from the existXCRDs() we would call `checkErr(err)` 
and `func checkErr(err error)` would do the if checks and print statements. 
maybe it doesn't make sense, not sure, just an idea, not a req

Reply:
No, that makes sense. That should be simple enough.

Reply:
added a common error process